### PR TITLE
README: Update notable users

### PR DESCRIPTION
**Pull request by jxs**
Updated and sorted the list with names of the projects and oficial description instead of just the url.
Removed the ones which don't seem to be using rust-libp2p anymore and added new users. 

Removed:
- https://github.com/paritytech/polkadot (couldn't find direct usages of rust-libp2p on the code)
- https://github.com/golemfactory/golem-libp2p (isn't updated since May 2019, couldn't find usage of nor golem-libp2p nor rust-libp2p on their new implementation [yagna](https://github.com/golemfactory/yagna))
- https://github.com/marcopoloprotocol/marcopolo (no longer exists, and I couldn't find usage of rust-libp2p in the [only Rust repo on the organization](https://github.com/mapprotocol/map-contracts))
-  https://www.actyx.com/developers/ (their [github organization](https://github.com/actyx-contrib) doesn't have any Rust repo)

Added:
- Iroh
- Locutus
- Subspace

Please feel free to correct if I erroneously removed any.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
> * [paritytech/polkadot](https://github.com/paritytech/polkadot) (couldn't find direct usages of rust-libp2p on the code)

Polkadot uses libp2p transitively through substrate but polkadot is probably the more known project. Not sure about this case :)

File: README.md [hunk 1]
```diff
@@ -83,14 +83,13 @@ Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
...
-- https://github.com/sigp/lighthouse
-- https://github.com/golemfactory/golem-libp2p
-- https://github.com/comit-network
-- https://github.com/rs-ipfs/rust-ipfs
-- https://github.com/marcopoloprotocol/marcopolo
-- https://github.com/ChainSafe/forest
-- https://github.com/ipfs-rust/ipfs-embed
-- https://www.actyx.com/developers/
-- https://github.com/starcoinorg/starcoin
+- [COMIT](https://github.com/comit-network/xmr-btc-swap) - An open protocol facilitating trustless cross-blockchain applications.
```
Comment (thomaseizinger) [inline README.md]: Disclaimer: I used to work on this.

The description is outdated in that the original project (comit-rs) is dead so it is a bit misleading. This project is also only community maintained but the description on that repo is more accurate: "Bitcoin–Monero Cross-chain Atomic Swap "

File: README.md [hunk 2]
```diff
@@ -83,14 +83,13 @@ Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
...
+- [Forest](https://github.com/ChainSafe/forest) - An implementation of Filecoin written in Rust.
+- [Lighthouse](https://github.com/sigp/lighthouse) - Ethereum consensus client in Rust.
+- [Locutus](https://github.com/freenet/locutus) - Global, observable, decentralized key-value store.
+- [Starcoin](https://github.com/starcoinorg/starcoin) - A smart contract blockchain network that scales by layering.
+- [Substrate](https://github.com/paritytech/substrate) - Framework for blockchain innovation.
+- [Subspace](https://github.com/subspace/subspace) - Subspace Network reference implementation
+- [ipfs-embed](https://github.com/ipfs-rust/ipfs-embed) - A small embeddable ipfs implementation.
+- [iroh](https://github.com/n0-computer/iroh) - Next-generation implementation of IPFS for Cloud & Mobile platforms.
+- [rust-ipfs](https://github.com/rs-ipfs/rust-ipfs) - IPFS implementation in Rust.
```
Comment (thomaseizinger) [inline README.md]: I think sorting this list alphabetically would be a good idea.

File: README.md (hunk 1 above)
Comment (jxs) [reply README.md]: Thanks Thomas, updated

File: README.md (hunk 2 above)
Comment (jxs) [reply README.md]: It is sorted alphabetically, you mean sorting alphabetically case insensitive?

**Comment by jxs** [comment]
> > * [paritytech/polkadot](https://github.com/paritytech/polkadot) (couldn't find direct usages of rust-libp2p on the code)
> 
> Polkadot uses libp2p transitively through substrate but polkadot is probably the more known project. Not sure about this case :)

good point, my rationale was to try to link to the code usages of `rust-libp2p`. 
Context: the idea came from the community call, where tutorials were being discussed and someone mentioned that usages of `rust-libp2p` could also serve as examples. 
But your question makes sense, what if we refer Polkadot on the Substrate description?

File: README.md [hunk 3]
```diff
@@ -83,14 +83,13 @@ Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
...
-- https://www.actyx.com/developers/
-- https://github.com/starcoinorg/starcoin
+- [COMIT](https://github.com/comit-network/xmr-btc-swap) - Bitcoin–Monero Cross-chain Atomic Swap.
+- [Forest](https://github.com/ChainSafe/forest) - An implementation of Filecoin written in Rust.
+- [Lighthouse](https://github.com/sigp/lighthouse) - Ethereum consensus client in Rust.
+- [Locutus](https://github.com/freenet/locutus) - Global, observable, decentralized key-value store.
+- [Starcoin](https://github.com/starcoinorg/starcoin) - A smart contract blockchain network that scales by layering.
+- [Subspace](https://github.com/subspace/subspace) - Subspace Network reference implementation
+- [Substrate](https://github.com/paritytech/substrate) - Framework for blockchain innovation.
+- [ipfs-embed](https://github.com/ipfs-rust/ipfs-embed) - A small embeddable ipfs implementation.
```
Comment (rkuhn) [inline README.md]: This is currently only driven by Actyx, it is just not in our github organisation (and the main Actyx codebase is private). Make of that what you want :-)

Comment (jxs) [reply README.md]: Hi Roland, thanks for the input. You mean `ipfs-embed` right? Would you like to add that it's maintained by Actyx with a link to your website?

File: README.md [hunk 4]
```diff
@@ -83,14 +83,13 @@ Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
...
-- https://github.com/ipfs-rust/ipfs-embed
-- https://www.actyx.com/developers/
-- https://github.com/starcoinorg/starcoin
+- [COMIT](https://github.com/comit-network/xmr-btc-swap) - Bitcoin–Monero Cross-chain Atomic Swap.
+- [Forest](https://github.com/ChainSafe/forest) - An implementation of Filecoin written in Rust.
+- [Lighthouse](https://github.com/sigp/lighthouse) - Ethereum consensus client in Rust.
+- [Locutus](https://github.com/freenet/locutus) - Global, observable, decentralized key-value store.
+- [Starcoin](https://github.com/starcoinorg/starcoin) - A smart contract blockchain network that scales by layering.
+- [Subspace](https://github.com/subspace/subspace) - Subspace Network reference implementation
+- [Substrate](https://github.com/paritytech/substrate) - Framework for blockchain innovation.
```
Comment (mxinden) [inline README.md]: Suggestion to reference Polkadot here makes sense to me.

File: README.md [hunk 5]
```diff
@@ -83,14 +83,13 @@ Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
...
-- https://github.com/rs-ipfs/rust-ipfs
-- https://github.com/marcopoloprotocol/marcopolo
-- https://github.com/ChainSafe/forest
-- https://github.com/ipfs-rust/ipfs-embed
-- https://www.actyx.com/developers/
-- https://github.com/starcoinorg/starcoin
+- [COMIT](https://github.com/comit-network/xmr-btc-swap) - Bitcoin–Monero Cross-chain Atomic Swap.
+- [Forest](https://github.com/ChainSafe/forest) - An implementation of Filecoin written in Rust.
+- [Lighthouse](https://github.com/sigp/lighthouse) - Ethereum consensus client in Rust.
+- [Locutus](https://github.com/freenet/locutus) - Global, observable, decentralized key-value store.
```
Comment (mxinden) [inline README.md]: //CC @sanity and @iduartgomez

**Review by mxinden — COMMENTED** [review]
Thanks @jxs!

I like the additional inline description.

File: README.md (hunk 4 above)
Comment (jxs) [reply README.md]: thanks Max, updated :)

File: README.md (hunk 3 above)
Comment (rkuhn) [reply README.md]: If you think it adds value to this list, then a simple “(used e.g. by Actyx)” on the ipfs-embed line would suffice as far as I’m concerned.

Comment (jxs) [reply README.md]: ok updated :)

File: README.md (hunk 2 above)
Comment (mxinden) [reply README.md]: Yes, I think sorting these case insensitive sounds good to me.

**Review by mxinden — APPROVED** [review]
Other than https://github.com/libp2p/rust-libp2p/pull/2922#discussion_r977910621 this looks good to me.

Thanks @jxs.

File: README.md [hunk 6]
```diff
@@ -83,14 +83,15 @@ Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
...
+- [Starcoin](https://github.com/starcoinorg/starcoin) - A smart contract blockchain network that scales by layering.
+- [Subspace](https://github.com/subspace/subspace) - Subspace Network reference implementation
+- [Substrate](https://github.com/paritytech/substrate) - Framework for blockchain innovation, used
+by [Polkadot](https://www.parity.io/technologies/polkadot/).
+- [ipfs-embed](https://github.com/ipfs-rust/ipfs-embed) - A small embeddable ipfs implementation
+used and maintained by [Actyx][https://www.actyx.com].
+- [iroh](https://github.com/n0-computer/iroh) - Next-generation implementation of IPFS for Cloud & Mobile platforms.
```
Comment (thomaseizinger) [inline README.md:90]: Not sure what alphabet you are using by `i` should be before `L` IMO :)

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks! Looks good modulo one comment!

File: README.md (hunk 2 above)
Comment (jxs) [reply README.md]: ok, updated :)

**Review by thomaseizinger — APPROVED** [review]
Thanks!

**Pull request merged**